d3a_interface/matching_algorithms/pay_as_clear_matching_algorithm.py

```diff
@@ -184,8 +184,8 @@ def _create_bid_offer_matches(
                     offers.insert(0, offer)
                     # Save the matching
                     bid_offer_matches.append(
-                        BidOfferMatch(market_id=market_id, bid=bid, selected_energy=bid_energy,
-                                      offer=offer, trade_rate=clearing_rate).serializable_dict()
+                        BidOfferMatch(market_id=market_id, bids=[bid], selected_energy=bid_energy,
```

Comment:
Nit: I think the content of this if-else can be refactored a bit:
```python
if offer_energy - bid_energy > MATCH_FLOATING_POINT_TOLERANCE:
    # Bid energy completely covered by offer energy
    # Update the residual offer energy to take into account the matched offer
    residual_offer_energy[offer["id"]] = offer_energy - bid_energy
    # Place the offer at the front of the offer list to cover following bids
    # since the offer still has some energy left
    offers.insert(0, offer)
    selected_energy = bid_energy
    # Update total clearing energy
    clearing_energy -= bid_energy
    # Set the bid energy to 0 to move forward to the next bid
    bid_energy = 0
else:
    # Offer is exhausted by the bid. More offers are needed to cover the bid.
    selected_energy = offer_energy
    # Subtract the offer energy from the bid, in order to not be taken into account
    # from following matches
    bid_energy -= offer_energy
    # Remove the offer from the residual offer dictionary
    residual_offer_energy.pop(offer["id"], None)
    # Update total clearing energy
    clearing_energy -= offer_energy

# Save the matching
bid_offer_matches.append(
    BidOfferMatch(
        market_id=market_id, bids=[bid], selected_energy=selected_energy,
        offers=[offer], trade_rate=clearing_rate).serializable_dict())
```